libraries/c_sdk/standard/mqtt/src/iot_mqtt_serializer_deserializer_wrapper.c

```diff
@@ -0,0 +1,645 @@
+    /*Deserializing PUBACK packet received from the netwok*/
+    mqttStatus = MQTT_DeserializeAck( &pIncomingPacket, &( pPuback->packetIdentifier ), NULL );
+    status = convertReturnCode( mqttStatus );
+    return status;
+}
+
+/*-----------------------------------------------------------*/
+
+/* Deserializer Ping Response wrapper .*/
+IotMqttError_t _deserializePingrespWrapper( _mqttPacket_t * pPingresp )
+{
+    IotMqttError_t status = IOT_MQTT_SUCCESS;
+    MQTTStatus_t mqttStatus = MQTTSuccess;
+    MQTTPacketInfo_t pIncomingPacket;
+
+    /* Null Check for Pingresponse packet  */
+    IotMqtt_Assert( pPingresp != NULL );
+
+    pIncomingPacket.type = pPingresp->type;
+    pIncomingPacket.pRemainingData = pPingresp->pRemainingData;
+    pIncomingPacket.remainingLength = pPingresp->remainingLength;
+    /*Deserializing PINGRESP packet received from the netwok*/
+    mqttStatus = MQTT_DeserializeAck( &pIncomingPacket, &( pPingresp->packetIdentifier ), NULL );
+    status = convertReturnCode( mqttStatus );
+    return status;
+}
+
+/*-----------------------------------------------------------*/
+
+/* Deserializer Publish wrapper .*/
+IotMqttError_t _deserializePublishWrapper( _mqttPacket_t * pPublish )
+{
+    IotMqttError_t status = IOT_MQTT_SUCCESS;
+    MQTTStatus_t mqttStatus = MQTTSuccess;
+    MQTTPacketInfo_t pIncomingPacket;
+    MQTTPublishInfo_t publishInfo;
+
+    /* Null Check for Publish packet  */
+    IotMqtt_Assert( pPublish != NULL );
+
+    pIncomingPacket.type = pPublish->type;
+    pIncomingPacket.pRemainingData = pPublish->pRemainingData;
+    pIncomingPacket.remainingLength = pPublish->remainingLength;
+
+
+    /*DeSerializing publish packet received from the netwok*/
+    mqttStatus = MQTT_DeserializePublish( &pIncomingPacket, &( pPublish->packetIdentifier ), &publishInfo );
+
+    status = convertReturnCode( mqttStatus );
+
+    if( status == IOT_MQTT_SUCCESS )
+    {
+        pPublish->u.pIncomingPublish->u.publish.publishInfo.qos = ( IotMqttQos_t ) publishInfo.qos;
+        pPublish->u.pIncomingPublish->u.publish.publishInfo.payloadLength = publishInfo.payloadLength;
+        pPublish->u.pIncomingPublish->u.publish.publishInfo.pPayload = publishInfo.pPayload;
+        pPublish->u.pIncomingPublish->u.publish.publishInfo.pTopicName = publishInfo.pTopicName;
+        pPublish->u.pIncomingPublish->u.publish.publishInfo.topicNameLength = publishInfo.topicNameLength;
+        pPublish->u.pIncomingPublish->u.publish.publishInfo.retain = publishInfo.retain;
+    }
+
+    return status;
+}
+
+/*-----------------------------------------------------------*/
+
+/* Suback Serializer Wrapper .*/
+IotMqttError_t _pubackSerializeWrapper( uint16_t packetIdentifier,
+                                        uint8_t ** pPubackPacket,
+                                        size_t * pPacketSize )
+{
+    IotMqttError_t status = IOT_MQTT_SUCCESS;
+    MQTTStatus_t mqttStatus = MQTTSuccess;
```

Comment:
It would be nice to have a better name to distinguish between these two statuses more easily.